app/services/katello/pulp/repository/yum.rb

```diff
@@ -55,6 +55,10 @@ def smart_proxy_download_policy
             policy
           end
         end
+
+        def regenerate_applicability
+          smart_proxy.pulp_extensions.repository.regenerate_applicability_by_ids(repo.pulp_id, true)
```

Comment:
I think you meam   `smart_proxy.pulp_api.extensions.repository.regenerate_applicability_by_ids`

this should fix your failing test +1 to the scenario tests :)